doc/getting_started/project_philosophy.rst

```diff
@@ -0,0 +1,73 @@
+Project Philosophy
+==================
+
+Many excellent plotting libraries exist in Python, including:
+
+* `Matplotlib <https://matplotlib.org/>`_
+* `Bokeh <https://bokeh.pydata.org/en/latest/>`_
+* `Seaborn <https://seaborn.pydata.org/>`_
+* `Lightning <http://lightning-viz.org>`_
+* `Plotly <https://plot.ly/>`_
+* `Pandas built-in plotting <https://pandas.pydata.org/pandas-docs/stable/visualization.html>`_
+* `HoloViews <https://holoviews.org>`_
+* `VisPy <https://vispy.org/>`_
+* `pygg <https://www.github.com/sirrice/pygg>`_
+
+Each library does a particular set of things well.
+
+User Challenges
+---------------
+
+However, such a proliferation of options creates great difficulty for users
+as they have to wade through all of these APIs to find which of them is the
+best for the task at hand. None of these libraries are optimized for
+high-level statistical visualization, so users have to assemble their own
+using a mishmash of APIs. For individuals just learning data science, this
+forces them to focus on learning APIs rather than exploring their data.
+
+Another challenge is current plotting APIs require the user to write code,
+even for incidental details of a visualization. This results in an unfortunate
+and unnecessary cognitive burden as the visualization type (histogram,
+scatterplot, etc.) can often be inferred using basic information such as the
+columns of interest and the data types of those columns.
+
+For example, if you are interested in the visualization of two numerical
+columns, a scatterplot is almost certainly a good starting point. If you add
+a categorical column to that, you probably want to encode that column using
+colors or facets. If inferring the visualization proves difficult at times, a
+simple user interface can construct a visualization without any coding.
+`Tableau <https://www.tableau.com/>`_ and the `Interactive Data
+Lab's <https://idl.cs.washington.edu/>`_
+`Polestar <https://github.com/vega/polestar>`_ and
+`Voyager <https://github.com/vega/voyager>`_ are excellent examples of such UIs.
+
+Design Approach and Solution
```

Comment:
I wrote[ a bit about this on SO](https://stackoverflow.com/questions/59845407/plotly-express-vs-altair-vega-lite-for-interactive-plots/66040502#66040502) a while back and many people seem to have found it helpful, so I wonder if it makes sense to include some parts of that answer here or maybe just a link. Another place that info could go is in a FAQ or the resource page, not sure where it fits the most naturally.

Reply:
Great comparison! I wasn't sure though how to best include its content here as this is more comparing Altair to most visualization libraries in general. I added a link to the SO answer so that it's easy to find and can serve as a deep-dive into the specific comparison of Altair vs. Plotly.

Reply:
Sounds like a good solution!